staging/src/k8s.io/apiserver/plugin/pkg/audit/dynamic/enforced/enforced.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package enforced
+
+import (
+	"fmt"
+
+	auditinternal "k8s.io/apiserver/pkg/apis/audit"
+	"k8s.io/apiserver/pkg/audit"
+	ev "k8s.io/apiserver/pkg/audit/event"
+	"k8s.io/apiserver/pkg/audit/policy"
+)
+
+// PluginName is the name reported in error metrics.
+const PluginName = "enforced"
+
+// Backend filters audit events according to the policy
+// trimming them as necessary to match the level
+type Backend struct {
+	policyChecker   policy.Checker
+	delegateBackend audit.Backend
+}
+
+// NewBackend returns an enforced audit backend that wraps delegate backend.
+// Enforced backend automatically runs and shuts down the delegate backend.
+func NewBackend(delegate audit.Backend, p policy.Checker) audit.Backend {
+	return &Backend{
+		policyChecker:   p,
+		delegateBackend: delegate,
+	}
+}
+
+// Run the delegate backend
+func (b Backend) Run(stopCh <-chan struct{}) error {
+	return b.delegateBackend.Run(stopCh)
+}
+
+// Shutdown the delegate backend
+func (b Backend) Shutdown() {
```

Comment:
same question about Shutdown() vs close(stopCh)

Reply:
in this scenario, the `stopCh` is cleanly propagated, with the dynamic backend things need to be handled differently because individual backends can be closed arbitrarily.